packages/pds/src/app-view/api/app/bsky/feed/getPostThread.ts

```diff
@@ -265,3 +271,76 @@ class ParentNotFoundError extends Error {
     super(`Parent not found: ${uri}`)
   }
 }
+
+// READ AFTER WRITE
+// ----------------
+
+const ensureReadAfterWrite = async (
+  ctx: AppContext,
+  requester: string,
+  res: ApiRes<OutputSchema>,
+): Promise<OutputSchema> => {
+  const clock = getClock(res.headers)
+  if (!clock) return res.data
+  if (!isThreadViewPost(res.data.thread)) {
+    return res.data
+  }
+  let thread: ThreadViewPost = res.data.thread
+  const localSrvc = ctx.services.local(ctx.db)
+  const local = await localSrvc.getRecordsSinceClock(requester, clock, [
+    ids.AppBskyFeedPost,
+  ])
+  const inThread = findPostsInThread(thread, local.posts)
+  if (inThread.length < 0) return res.data
+  for (const record of inThread) {
+    thread = await insertIntoThreadReplies(localSrvc, thread, record)
+  }
+  return {
+    ...res.data,
+    thread,
+  }
+}
+
+const findPostsInThread = (
+  thread: ThreadViewPost,
+  posts: RecordDescript<PostRecord>[],
+): RecordDescript<PostRecord>[] => {
+  return posts.filter((post) => {
+    const rootUri = post.record.reply?.root.uri
+    if (!rootUri) return false
+    if (rootUri === thread.post.uri) return true
+    return (thread.post.record as PostRecord).reply?.root.uri === rootUri
+  })
+}
+
+const insertIntoThreadReplies = async (
+  localSrvc: LocalService,
+  view: ThreadViewPost,
+  descript: RecordDescript<PostRecord>,
+): Promise<ThreadViewPost> => {
+  if (descript.record.reply?.parent.uri === view.post.uri) {
+    const postView = await localSrvc.getPost(descript)
+    if (!postView) return view
+    const threadPostView = {
+      $type: 'app.bsky.feed.defs#threadViewPost',
+      post: postView,
+    }
+    const replies = [threadPostView, ...(view.replies ?? [])]
+    return {
+      ...view,
+      replies,
+    }
+  }
+  if (!view.replies) return view
+  const replies = await Promise.all(
+    view.replies.map(async (reply) =>
+      isThreadViewPost(reply)
+        ? await insertIntoThreadReplies(localSrvc, reply, descript)
+        : reply,
+    ),
+  )
```

Comment:
hmmm i feel fairly comfortable with this & generally it won't even _try_ to do anything deeper than one, because the app will just reload the immediate parent

maybe woulda been worth it if I'd thought to do that before writing this, but now it's good to go 😛